packages/runtime-common/ai/prompt.ts

```diff
@@ -1067,14 +1072,197 @@ function formatCorrectnessResultSummary(
   };
 }
 
+function findCheckCorrectnessCommandRequest(
+  history: DiscreteMatrixEvent[],
+  commandRequestId: string,
+): CommandRequest | undefined {
+  for (let event of history) {
+    let requests = getCheckCorrectnessCommandRequests(event);
+    let match = requests.find((request) => request.id === commandRequestId);
+    if (match) {
+      return match;
+    }
+  }
+  return undefined;
+}
+
+function formatCorrectnessTargetKeyWithEvent(
+  targetType?: string,
+  targetRef?: string,
+  targetEventId?: string,
+): string | undefined {
+  if (!targetRef) {
+    return undefined;
+  }
+  let normalizedType = targetType ?? 'target';
+  let eventPart = targetEventId ? `|event:${targetEventId}` : '';
+  return `${normalizedType}:${String(targetRef)}${eventPart}`;
+}
+
+type CheckCorrectnessTargetParts = {
+  targetType?: string;
+  targetRef?: string;
+  targetEventId?: string;
+};
+
+function extractCheckCorrectnessTargetParts(
+  request?: CommandRequest,
+): CheckCorrectnessTargetParts {
+  if (!request) {
+    return {};
+  }
+  let args = (request.arguments as Record<string, any>) ?? {};
+  let attributes = (args.attributes as Record<string, any>) ?? {};
+  let targetRef =
+    attributes.targetRef ??
+    args.targetRef ??
+    attributes.fileUrl ??
+    args.fileUrl ??
+    attributes.cardId ??
+    args.cardId;
+  if (!targetRef) {
+    return {};
+  }
+  let targetType =
+    attributes.targetType ??
+    args.targetType ??
+    attributes.target_type ??
+    args.target_type;
+  if (!targetType) {
+    if (attributes.cardId || args.cardId) {
+      targetType = 'card';
+    } else if (attributes.fileUrl || args.fileUrl) {
+      targetType = 'file';
+    }
+  }
+  let targetEventId =
+    attributes.targetEventId ??
+    args.targetEventId ??
+    attributes.target_event_id ??
+    args.target_event_id ??
+    undefined;
```

Comment:
Why do we have to support so many permutations?

Reply:
It was for backwards compatibility reasons but it turns out we don't really need it. I made the adjustments in next commits and simplified `CheckCorrectnessInput` by getting rid of `cardId` and `fileUrl`, and only keep the `targetType` and `targetRef`.